src/screens/FilterScreen.js

```diff
@@ -1,34 +1,198 @@
 import React, {Component} from 'react';
-import {Text} from 'react-native';
+import {Text, ScrollView, View, StyleSheet, Platform} from 'react-native';
+import {connect} from "react-redux";
 
+import _ from 'lodash';
 
+import {CheckBox, Button, Icon} from 'react-native-elements';
+import Accordion from 'react-native-collapsible/Accordion';
 
-import {CardSection, Card} from '../components/common/index';
-import MapScreen from "../components/SearchMap";
+import {checkboxWasClicked} from "../actions";
+
+import {navKeys} from '../constants';
+import {campsite, more_screen, reducerAlerts} from '../locale.en';
+import {campsiteIcon, grey} from "../styles";
+
+const {campsite_form: {accessibility, facilities, price, accessibility_options, facilities_options, price_options}} = campsite;
+
+
+const ACCESSIBILITY = [
+    {
+        title: accessibility,
+        content: accessibility_options
+    }
+];
+
+const FACILITIES = [
+    {
+        title: facilities,
+        content: facilities_options
+    }
+];
+
+const PRICE = [
+    {
+        title: price,
+        content: price_options
+    }
+];
 
 class FilterScreen extends Component {
 
+    componentWillMount() {
+        const {displaySites} = this.props;
+
+        this.props.navigation.setParams({siteCount: displaySites.length});
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const nextSetOfSites = nextProps.displaySites;
+        const currentSites = this.props.displaySites;
+
+
+        if (nextSetOfSites.length !== currentSites.length) {
+            this.props.navigation.setParams({siteCount: nextSetOfSites.length});
+        }
```

Comment:
@amgoodman5 I learned a good trick here... if we do `this.props.navigation.setParams`, we'll be triggering new props, which will fire `componentWillReceiveProps` again, which fires off the `setParams` again, and drops us into an infinite loop. But we do want to make sure that when the props change for the site counts that we can update the top right nav results dynamically. What I did here was a way to skip the infinite loop where we call `setParams` (which will trigger another `componentWillReceiveProps`, but if the site count coming in is the same as what we had, we just skip and don't go into another cycle. This way we only trigger the cycle again when we actually have something worth changing.

Reply:
👍